packages/unmock-hash/src/hash/v0/index.ts

```diff
@@ -188,7 +188,17 @@ export default (
   initialInput: IHashableV0,
   ignore?: IgnoreV0,
   action?: ActionsV0 | ActionsV0[],
-) =>
-  objectHash(
-    applyActions(makeIgnorable(initialInput, ignore || []), action || []),
-  ).substring(0, TRUNCATE_HASH_AT);
+  challenge?: string,
+  onError?: (
+    initialInput: IHashableV0,
+    afterIgnore: Partial<IHashableV0>,
+    afterActions: ISerializedHashableWithIgnoreV0) => void,
+) => {
```

Comment:
I think with the proliferation of arguments here, it might make sense to include them as a single object so one does not have to pass `undefined`, and enforce the use of argument names.

I'm guessing that `challenge` is meant to be e.g. the hash computed by the cloud? It would be helpful to add a  comment describing it.